src/move/rewards/sources/emojicoin_dot_fun_arena.move

```diff
@@ -0,0 +1,614 @@
+module rewards::emojicoin_dot_fun_arena {
+
+    use aptos_framework::account::{Self, SignerCapability};
+    use aptos_framework::aptos_account;
+    use aptos_framework::aptos_coin::AptosCoin;
+    use aptos_framework::coin::{Self, Coin};
+    use aptos_framework::randomness::Self;
+    use aptos_framework::timestamp;
+    use aptos_std::smart_table::{Self, SmartTable};
+    use emojicoin_dot_fun::emojicoin_dot_fun::{
+        Self,
+        MarketMetadata,
+        market_view,
+        registry_view,
+        unpack_market_metadata,
+        unpack_registry_view
+    };
+    use std::option::Self;
+    use std::signer;
+
+    /// Signer does not correspond to rewards account.
+    const E_NOT_REWARDS: u64 = 0;
+
+    /// Resource account address seed for the registry.
+    const REGISTRY_SEED: vector<u8> = b"Arena registry";
+
+    const U64_MAX: u64 = 0xffffffffffffffff;
+
+    const LOCK_IN_PERIOD_HOURS: u64 = 12;
+    const MELEE_DURATION_HOURS: u64 = 36;
+    const MICROSECONDS_PER_HOUR: u64 = 3_600_000_000;
+
+    /// Flat integrator fee.
+    const INTEGRATOR_FEE_RATE_BPS: u8 = 100;
+    const INTEGRATOR_FEE_RATE_BPS_DUAL_ROUTE: u8 = 50;
+    const REWARDS_PER_MELEE: u64 = 1500 * 100_000_000;
+    const MATCH_PERCENTAGE: u64 = 20;
+    const MAX_PERCENTAGE: u64 = 100;
+    const MAX_MATCH_AMOUNT: u64 = 100_000_000;
+
+    struct Melee has copy, drop, store {
+        /// 1-indexed for conformity with market ID indexing.
+        melee_id: u64,
+        /// Market with lower market ID first.
+        market_metadatas: vector<MarketMetadata>,
+        available_rewards_in_octas: u64,
+        /// In microseconds.
+        start_time: u64,
+        lock_in_period: u64,
+        duration: u64
+    }
+
+    struct Nil has store {}
+
+    struct MeleeEscrow<phantom Emojicoin0, phantom EmojicoinLP0, phantom Emojicoin1, phantom EmojicoinLP1> has key {
```

Comment:
Adding a field like `octa_holdings_upon_entry` could enable instantaneous profit and loss calculations, however it would need to be reset any time someone decided to deposit more. Otherwise there would need to be a sequence of `_growth_q64` values to determine PnL between successive entry/exit operations

Reply:
Resolving per `MeleeEscrow.octas_entered`